core/include/detray/utils/tuple.hpp

```diff
@@ -25,12 +25,18 @@ struct tuple<T, Ts...> {
     DETRAY_HOST_DEVICE constexpr tuple(){};
 
     template <typename U, typename... Us>
-    requires std::is_constructible_v<T, U &&>
-        &&std::is_constructible_v<tuple<Ts...>, Us &&...>
+    requires std::is_constructible_v<T, const U &>
+        &&std::is_constructible_v<tuple<Ts...>, const Us &...>
```

Comment:
I guess it's obvious to you as well by now, but the types of these should not change just because we add a move constructor as well. We still need to use a "generic template type" here. 🤔

Reply:
No but check out the definition of this function, the types through which the tuple are constructed are const references; I guess I just made a mistake in the original definition of the constructor.

Reply:
Indeed, past me was smarter than present-day me here. :smile: